content/en/blogs/devops-automatic-testing.md

```diff
@@ -0,0 +1,247 @@
+---
+title: 'Use KubeSphere DevOps to Build an Automated Testing System'
+tag: 'KubeSphere,Kubernetes'
+createTime: '2020-04-29'
+author: 'Shaowen Chen'
+snapshot: 'https://pek3b.qingstor.com/kubesphere-docs/png/20200428175258.png'
+---
+
+## Layers of Testing
+
+The purpose of testing is to verify expected functionality and find potential defects. Testing enhances the confidence in delivering qualified products and also brings the possibility of agile iteration. It is safe to say that testing determines the product development progress.
+
+The network model includes seven layers of OSI model and four layers of TCP model, and the development modes include MTV, MVC, MVP, MVVM, etc. These technologies gradually mature as they grow from high cohesion, low coupling, and division of responsibilities, modules, and layers to structuralization and standardization.
+
+Testing is also divided into UI testing, API testing, and unit testing. Testing is a way to balance output and cost rather than a new type of technology.
```

Comment:
->
Likewise, we also have different types of testing - UI testing, API testing, and unit testing. Testing is a way to balance output and cost rather than a new type of technology.